worker/buildbot_worker/commands/transfer.py

```diff
@@ -224,7 +224,11 @@ def start(self):
         # Not possible with older versions:
         # exceptions.AttributeError: 'TarFile' object has no attribute '__exit__'
         archive = tarfile.open(mode=mode, fileobj=self.fp)
-        archive.add(self.path, '')
+        try:
+            archive.add(self.path, '')
+        except OSError:
+            print("Error:{} is not a valid directory".format(self.path))
```

Comment:
same, but in that case, you would use

```suggestion
           self.sendStatus({'header': "Error:{} is not a valid directory".format(self.path)})
```